packages/rulesets/src/spectral/functions/validate-patch-body-param-properties.ts

```diff
@@ -55,6 +61,36 @@ export const validatePatchBodyParamProperties = createRulesetFunction<unknown, O
       if (_opts.shouldNot) {
         _opts.shouldNot.forEach((p: string) => {
           const property = getProperties(bodyParameter)?.[p]
+
+          if (property) {
+            let isPropertyReadOnly = false
+            let isPropertyImmutable = false
+
+            if (property["readOnly"] && property["readOnly"] === true) {
+              isPropertyReadOnly = true
+            }
+
+            if (property["x-ms-mutability"] && Array.isArray(property["x-ms-mutability"])) {
+              const schemaArray = property["x-ms-mutability"]
+              if (!schemaArray.includes("update")) {
+                isPropertyImmutable = true
+              }
+            }
+
+            if (!(isPropertyReadOnly || isPropertyImmutable)) {
+              errors.push({
+                message: `Mark the top-level property "${p}", specified in the patch operation body, as readOnly or immutable. You could also choose to remove it from the request payload of the Patch operation. This property is not patchable.`,
+                path: [...path, "parameters", index],
+              })
+            }
+          }
+        })
+      }
+      if (_opts.propertiesThatMustNotBeInPropertiesBagAsWritable) {
+        _opts.propertiesThatMustNotBeInPropertiesBagAsWritable.forEach((p: string) => {
+          const propertiesProperty = getProperties(bodyParameter)?.["properties"]
```

Comment:
>getProperties(bodyParameter) [](http://example.com/codeflow?start=37&length=28)

Instead of fetching these properties in every block, can we put in a common place?

Reply:
There are slight differences in error msg etc. so preferred to leave it this way